src/main/java/duke/commands/ListCommand.java

```diff
@@ -0,0 +1,369 @@
+                                if (note.getBudgetObject() != null) {
+                                    notes.add(note);
+                                    selectionSortBudgets(notes);
+                                }
+                            } else {
+                                switch (NoteType.getCONSTRUCTOR(this.noteType.toString())) {
+                                    case "Bill" -> {
+                                        if (note instanceof Bill) {
+                                            notes.add(note);
+                                            selectionSortDates(notes);
+                                        }
+                                    }
+                                    case "Birthday" -> {
+                                        if (note instanceof Birthday) {
+                                            notes.add(note);
+                                            selectionSortDates(notes);
+                                        }
+                                    }
+                                    case "Deadline" -> {
+                                        if (note instanceof Deadline) {
+                                            notes.add(note);
+                                            selectionSortDates(notes);
+                                        }
+                                    }
+                                    case "Event" -> {
+                                        if (note instanceof Event) {
+                                            notes.add(note);
+                                            selectionSortDates(notes);
+                                        }
+                                    }
+                                    case "Shoplist" -> {
+                                        if (note instanceof Shoplist) {
+                                            notes.add(note);
+                                            selectionSortBudgets(notes);
+                                        }
+                                    }
+                                    case "Task" -> {
+                                        if (note instanceof Task) {
+                                            notes.add(note);
+                                        }
+                                    }
+                                    case "Wedding" -> {
+                                        if (note instanceof Wedding) {
+                                            notes.add(note);
+                                            selectionSortDates(notes);
+                                        }
+                                    }
+                                    case "Note" -> notes.add(note);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Way too deep nesting. consider extracting the body of the `for` loop into a separate method. Inside that method, use guard clauses to reduce nesting.

Reply:
Thanks Prof, I've changed the if-statements to guard clauses to reduce the nesting.